esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -851,28 +949,44 @@ EsdlBindingImpl::EsdlBindingImpl()
 {
 }
 
-EsdlBindingImpl::EsdlBindingImpl(IPropertyTree* cfg,
-                                 const char *binding,
-                                 const char *process) : CHttpSoapBinding(cfg, binding, process)
+EsdlBindingImpl::EsdlBindingImpl(IPropertyTree* cfg, const char *binding,  const char *process) : CHttpSoapBinding(cfg, binding, process)
 {
     m_bindingName.set(binding);
     m_processName.set(process);
 
     try
     {
-        DBGLOG("ESDL Binding %s is subscribing to all /ESDL/Bindings/Binding dali changes", binding);
-        //Since it seems we cannot subscribe to a specific /ESDL/Bindings/Binding, all bindings have to subscribe to the /ESDL/Bindings branch
-        m_pBindingSubscription.clear();
-        m_pBindingSubscription.setown( new CESDLBindingSubscription(this) );
+        char currentDirectory[_MAX_DIR];
+        if (!getcwd(currentDirectory, sizeof(currentDirectory)))
+            throw makeWsException( ERR_ESDL_BINDING_UNEXPECTED, WSERR_CLIENT,  "ESP", "getcwd failed");
 
-        DBGLOG("ESDL Binding %s is subscribing to all /ESDL/Bindings/Definition dali changes", binding);
-        //Since it seems we cannot subscribe to a specific /ESDL/Definitions/Definition, all bindings have to subscribe to the /ESDL/Definitions branch
-        m_pDefinitionSubscription.clear();
-        m_pDefinitionSubscription.setown( new CESDLDefinitionSubscription(this) );
+        m_esdlStateFilesLocation.setf("%s%cDESDL_STATE",currentDirectory, PATHSEPCHAR);
+        recursiveCreateDirectory(m_esdlStateFilesLocation.str());
 
-        m_esdlBndCfg.set(fetchESDLBindingFromDali(process, binding));
+        m_esdlStateFilesLocation.appendf("%c%s-%s.xml", PATHSEPCHAR, m_processName.str(), m_bindingName.str());
```

Comment:
Good this should make the name unique even if running multiple ESDL bindings on the same process.